src/rules/selector-complexity.js

```diff
@@ -0,0 +1,479 @@
+					maxIds: {
+						type: "integer",
+						minimum: 0,
+					},
+					maxClasses: {
+						type: "integer",
+						minimum: 0,
+					},
+					maxTypes: {
+						type: "integer",
+						minimum: 0,
+					},
+					maxAttributes: {
+						type: "integer",
+						minimum: 0,
+					},
+					maxPseudoClasses: {
+						type: "integer",
+						minimum: 0,
+					},
+					maxUniversals: {
+						type: "integer",
+						minimum: 0,
+					},
+					maxCompounds: {
+						type: "integer",
+						minimum: 0,
+					},
+					maxCombinators: {
+						type: "integer",
+						minimum: 0,
+					},
+					disallowCombinators: {
+						type: "array",
+						items: {
+							type: "string",
+						},
+						uniqueItems: true,
+					},
+					disallowPseudoClasses: {
+						type: "array",
+						items: {
+							type: "string",
+						},
+						uniqueItems: true,
+					},
+					disallowPseudoElements: {
+						type: "array",
+						items: {
+							type: "string",
+						},
+						uniqueItems: true,
+					},
+					disallowAttributes: {
+						type: "array",
+						items: {
+							type: "string",
+						},
+						uniqueItems: true,
+					},
+					disallowAttributeMatchers: {
+						type: "array",
+						items: {
+							type: "string",
+						},
+						uniqueItems: true,
+					},
+				},
+			},
+		],
+
+		defaultOptions: [
+			{
+				maxIds: Infinity,
+				maxClasses: Infinity,
+				maxTypes: Infinity,
+				maxAttributes: Infinity,
+				maxPseudoClasses: Infinity,
+				maxUniversals: Infinity,
+				maxCompounds: Infinity,
+				maxCombinators: Infinity,
+				disallowCombinators: [],
+				disallowPseudoClasses: [],
+				disallowPseudoElements: [],
+				disallowAttributes: [],
+				disallowAttributeMatchers: [],
+			},
+		],
+
+		messages: {
+			maxSelectors:
+				"Exceeded maximum {{selector}} selector. Only {{limit}} allowed.",
+			disallowedSelectors:
+				"'{{selectorName}}' {{selector}} is not allowed.",
+		},
+	},
+
+	create(context) {
+		const [
+			{
+				maxIds,
+				maxClasses,
+				maxTypes,
+				maxAttributes,
+				maxPseudoClasses,
+				maxUniversals,
+				maxCompounds,
+				maxCombinators,
+				disallowCombinators,
+				disallowPseudoClasses,
+				disallowPseudoElements,
+				disallowAttributes,
+				disallowAttributeMatchers,
+			},
+		] = context.options;
+
+		return {
+			Selector(node) {
+				const selectors = node.children;
+				const selectorLoc = node.loc;
+
+				const idSelectors = getSelectors(selectors, "IdSelector");
+				const classSelectors = getSelectors(selectors, "ClassSelector");
+				const typeSelectors = selectors.filter(
+					child =>
+						child.type === "TypeSelector" && child.name !== "*",
+				);
+				const attributeSelectors = getSelectors(
+					selectors,
+					"AttributeSelector",
+				);
+				const pseudoClassSelectors = getSelectors(
+					selectors,
+					"PseudoClassSelector",
+				);
+				const universalSelectors = selectors.filter(
+					child =>
+						child.type === "TypeSelector" && child.name === "*",
+				);
```

Comment:
Conversely, we could add a test to ensure that regular `TypeSelector`s aren't counted as universals.